controller/array_action/array_mediator_ds8k.py

```diff
@@ -52,8 +52,15 @@ def parse_version(bundle):
     return '.'.join([v1[-1], v2[0], v2[1]])
 
 
-def get_volume_id_from_scsi_identifier(scsi_id):
-    return scsi_id[-4:]
+def get_volume_id_from_scsi_identifier(func):
```

Comment:
nice!
here is how I would write it:
```
from decorator import decorator

VOLUME_ID_PARAM_NAME = 'volume_id'

def scsi_id_to_volume_id(scsi_id):
    return scsi_id[-4:]

def try_convert_first_arg(converter, args):
    if args:
        converted = converter(args[0])
        return (converted,) + args[1:]
    return ()

def try_convert_arg_by_name(converter, name, kwargs):
    if name in kwargs:
        kwargs[name] = converter(kwargs[name])

@decorator
def convert_volume_id_from_scsi_id(func, *args, **kwargs):
    args = try_convert_first_arg(scsi_id_to_volume_id, args)
    try_convert_arg_by_name(scsi_id_to_volume_id, VOLUME_ID_PARAM_NAME, kwargs)
    return func(*args, **kwargs)

@convert_volume_id_from_scsi_id
def is_flashcopy_source(volume_id, volume_flashcopy):
    ....
```
feel free to take whatever you like from it

Reply:
but its not always "volume_id"

Reply:
right. then consider this:
```
def convert_from_scsi_id(scsi_id_param_name):
    @decorator
    def convert_to_object_id(func, self, *args, **kwargs):
        args = try_convert_first_arg(scsi_id_to_volume_id, args)
        if not args:
            try_convert_arg_by_name(scsi_id_to_volume_id, scsi_id_param_name, kwargs)
        return func(self, *args, **kwargs)
    return convert_to_object_id

@convert_from_scsi_id('volume_id')
def delete_volume(self, volume_id):
    ....
```

Reply:
done

Reply:
just don't forget to cover the kwargs use case in the unit tests as well